packages/core/src/modules/credentials/protocol/v2/__tests__/v2.ldproof.credentials-auto-accept.test.ts

```diff
@@ -0,0 +1,481 @@
+        state: CredentialState.Done,
+      })
+    })
+    test('Faber starts with V2 credential offer to Alice, both with autoAcceptCredential on `contentApproved`', async () => {
+      testLogger.test('Faber sends credential offer to Alice')
+      const offerOptions: OfferCredentialOptions = {
+        comment: 'some comment about credential',
+        connectionId: faberConnection.id,
+        credentialFormats: {
+          jsonld: signCredentialOptions,
+        },
+        protocolVersion: CredentialProtocolVersion.V2,
+      }
+      const faberCredentialExchangeRecord = await faberAgent.credentials.offerCredential(offerOptions)
+
+      testLogger.test('Alice waits for credential offer from Faber')
+      aliceCredentialRecord = await waitForCredentialRecord(aliceAgent, {
+        threadId: faberCredentialExchangeRecord.threadId,
+        state: CredentialState.OfferReceived,
+      })
+
+      // below values are not in json object
+      expect(aliceCredentialRecord.id).not.toBeNull()
+      expect(aliceCredentialRecord.getTags()).toEqual({
+        threadId: aliceCredentialRecord.threadId,
+        state: aliceCredentialRecord.state,
+        connectionId: aliceConnection.id,
+        credentialIds: [],
+      })
+      expect(aliceCredentialRecord.type).toBe(CredentialExchangeRecord.type)
+
+      if (aliceCredentialRecord.connectionId) {
+        // we do not need to specify connection id in this object
+        // it is either connectionless or included in the offer message
+        const acceptOfferOptions: AcceptOfferOptions = {
+          credentialRecordId: aliceCredentialRecord.id,
+        }
+        testLogger.test('Alice sends credential request to faber')
+        const faberCredentialExchangeRecord: CredentialExchangeRecord = await aliceAgent.credentials.acceptOffer(
+          acceptOfferOptions
+        )
+
+        testLogger.test('Alice waits for credential from Faber')
+        aliceCredentialRecord = await waitForCredentialRecord(aliceAgent, {
+          threadId: faberCredentialExchangeRecord.threadId,
+          state: CredentialState.CredentialReceived,
+        })
+
+        testLogger.test('Faber waits for credential ack from Alice')
+
+        const faberCredentialRecord = await waitForCredentialRecord(faberAgent, {
+          threadId: faberCredentialExchangeRecord.threadId,
+          state: CredentialState.Done,
+        })
+
+        expect(aliceCredentialRecord).toMatchObject({
+          type: CredentialExchangeRecord.type,
+          id: expect.any(String),
+          createdAt: expect.any(Date),
+          metadata: {},
+          state: CredentialState.Done,
+        })
+
+        expect(faberCredentialRecord).toMatchObject({
+          type: CredentialExchangeRecord.type,
+          id: expect.any(String),
+          createdAt: expect.any(Date),
+          state: CredentialState.Done,
+        })
+      } else {
+        throw new AriesFrameworkError('missing alice connection id')
+      }
```

Comment:
If we just do the throw in the if, we don't need the else (and have less nesting)

```suggestion
      if (!aliceCredentialRecord.connectionId)  throw new AriesFrameworkError('missing alice connection id')
      
	    // we do not need to specify connection id in this object
	    // it is either connectionless or included in the offer message
	    const acceptOfferOptions: AcceptOfferOptions = {
	    credentialRecordId: aliceCredentialRecord.id,
	    }
	    testLogger.test('Alice sends credential request to faber')
	    const faberCredentialExchangeRecord: CredentialExchangeRecord = await aliceAgent.credentials.acceptOffer(
	    acceptOfferOptions
	    )
	    
	    testLogger.test('Alice waits for credential from Faber')
	    aliceCredentialRecord = await waitForCredentialRecord(aliceAgent, {
	    threadId: faberCredentialExchangeRecord.threadId,
	    state: CredentialState.CredentialReceived,
	    })
	    
	    testLogger.test('Faber waits for credential ack from Alice')
	    
	    const faberCredentialRecord = await waitForCredentialRecord(faberAgent, {
	    threadId: faberCredentialExchangeRecord.threadId,
	    state: CredentialState.Done,
	    })
	    
	    expect(aliceCredentialRecord).toMatchObject({
	    type: CredentialExchangeRecord.type,
	    id: expect.any(String),
	    createdAt: expect.any(Date),
	    metadata: {},
	    state: CredentialState.Done,
	    })
	    
	    expect(faberCredentialRecord).toMatchObject({
	    type: CredentialExchangeRecord.type,
	    id: expect.any(String),
	    createdAt: expect.any(Date),
	    state: CredentialState.Done,
	    })
```